### PR TITLE
Explicit jsonb support for custom pg clients

### PR DESCRIPTION
**Pull request by viqueen**
I ran into an issue where my `jsonb` columns are created as `text` when using `pg-mem` in tests and aurora friendly clients like `knex-data-api-client`

It resulted in a lot of `cast blah as jsonb` workarounds, but really, the issue is that under the hood, knex just falls back to `text` data type when it cannot resolve a valid pg version.

This change lets custom clients like `knex-data-api` explicitly state they support `jsonb` instead of passing in a potentially wrong postgres version (think serverless setup)

### CODE REVIEW

**Comment by OlivierCavadenti** [comment]
Ready for review ?

**Comment by viqueen** [comment]
@OlivierCavadenti it's ready , though I think I should leave the check on `cockroachdb` client in place to avoid breaking somebody's world

**Comment by viqueen** [comment]
@kibertoad documentation added over here https://github.com/knex/documentation/pull/423
I added a unit test only.
a meaningful integration test would be with a serverless db like aurora postgres, I'm thinking to wire it up with localstack, but can't sink more time this week.
unit test sounds like a good middle ground for now , what do you think ?

**Comment by viqueen** [comment]
and another ping @kibertoad , is there more to do documentation wise ?
I've already opened this https://github.com/knex/documentation/pull/423 , but it's also not reviewed yet

**Comment by kibertoad** [comment]
sorry for the delay, I was swamped a bit
will take a look over the weekend

**Comment by kibertoad** [comment]
> unit test sounds like a good middle ground for now , what do you think ?
sure, let's go with a unit test

File: lib/dialects/postgres/schema/pg-columncompiler.js [hunk 1]
```diff
@@ -139,7 +139,8 @@ function jsonColumn(client, jsonb) {
   if (
     !client.version ||
     client.config.client === 'cockroachdb' ||
-    parseFloat(client.version) >= 9.2
+    parseFloat(client.version) >= 9.2 ||
+    client.config.jsonbSupport === true
```
Comment (kibertoad) [inline lib/dialects/postgres/schema/pg-columncompiler.js:143]:
```suggestion
    client.config.jsonbSupport === true ||
    parseFloat(client.version) >= 9.2
```

File: test/integration2/util/knex-instance-provider.js [hunk 2]
```diff
@@ -132,6 +132,7 @@ const testConfigs = {
     pool,
     migrations,
     seeds,
+    jsonbSupport: true,
```
Comment (kibertoad) [inline test/integration2/util/knex-instance-provider.js:135]:
```suggestion
```

**Review by kibertoad — APPROVED** [review]

**Comment by kibertoad** [comment]
Released in 2.3.0

**Pull request merged**